src/lib.rs

```diff
@@ -885,51 +925,747 @@ impl GooseAttack {
+        }
+
+        Ok(())
+    }
+
+    // Configure maximum run time if specified, otherwise run until canceled.
+    fn set_run_time(&mut self) -> Result<(), GooseError> {
+        // Use --run-time if set, don't allow on Worker.
+        self.run_time = if !self.configuration.run_time.is_empty() {
+            if self.attack_mode == GooseMode::Worker {
+                return Err(GooseError::InvalidOption {
+                    option: "--run-time".to_string(),
+                    value: "true".to_string(),
+                    detail: "The --run-time option can not be set together with the --worker flag."
+                        .to_string(),
+                });
+            }
+            util::parse_timespan(&self.configuration.run_time)
+        // Otherwise, use default if set, but not on Worker.
+        } else if let Some(r) = self.defaults.run_time {
+            if self.attack_mode == GooseMode::Worker {
+                0
+            } else {
+                r
+            }
+        }
+        // Otherwise the test runs until canceled.
+        else {
+            0
+        };
+
+        if self.run_time > 0 {
+            info!("run_time = {}", self.run_time);
+        }
+
+        Ok(())
+    }
+
+    // Configure how quickly to hatch Goose Users.
+    fn set_hatch_rate(&mut self) -> Result<(), GooseError> {
+        self.hatch_rate = if self.configuration.hatch_rate > 0 {
+            // Don't allow --hatch-rate with --worker.
+            if self.attack_mode == GooseMode::Worker {
+                return Err(GooseError::InvalidOption {
+                    option: "--hatch-rate".to_string(),
+                    value: self.configuration.hatch_rate.to_string(),
+                    detail:
+                        "The --hatch-rate option can not be set together with the --worker flag."
+                            .to_string(),
+                });
+            }
+            self.configuration.hatch_rate
+        } else if let Some(hatch_rate) = self.defaults.hatch_rate {
+            if self.attack_mode == GooseMode::Worker {
+                self.hatch_rate
+            } else {
+                hatch_rate
+            }
+        } else {
+            self.hatch_rate
+        };
+
+        // If --hatch-rate isn't set, use default if set and not in Worker mode.
+        if self.hatch_rate == 0 && self.attack_mode != GooseMode::Worker {
+            return Err(GooseError::InvalidOption {
+                option: "--hatch-rate".to_string(),
+                value: self.configuration.hatch_rate.to_string(),
+                detail: "The --hatch-rate option must be set to at least 1.".to_string(),
+            });
+        }
+
+        if self.hatch_rate > 0 {
+            info!("hatch_rate = {}", self.configuration.hatch_rate);
+        }
+
+        Ok(())
+    }
+
+    // Configure maximum requests per second if throttle enabled.
+    fn set_throttle_requests(&mut self) -> Result<(), GooseError> {
```

Comment:
@LionsAd Another that's going into `GooseAttack` but could just overload `configuration`.

Reply:
I re-removed this from `GooseAttack` and manage it entirely in `configuration`.